tendermint/src/merkle.rs

```diff
@@ -8,16 +8,21 @@ pub const HASH_SIZE: usize = 32;
 /// Hash is the output of the cryptographic digest function
 pub type Hash = [u8; HASH_SIZE];
 
-/// Compute a simple Merkle root from the arbitrary sized byte slices
-pub fn simple_hash_from_byte_slices(byte_slices: &[&[u8]]) -> Hash {
+/// Compute a simple Merkle root from the arbitrary byte arrays
+pub fn simple_hash_from_byte_slices(byte_slices: Vec<Vec<u8>>) -> Hash {
```

Comment:
Those aren't slices 😉

Reply:
Changed to `simple_hash_from_byte_vectors`  in 937ba67.

I think a better name for the the method  would rather be `compute_merkle_root_from_leaves` or `compute_merkle_root_from_leaf_values` instead of `simple_hash_from_[some_type]`. 
This would actually say what the method does instead of what type it expects as input (which the method signature clearly states). What do you think? CC @ebuchman

Reply:
You're right about better naming happy to see this changed.